src/Microsoft.DotNet.Build.Tasks/PackageFiles/buildagainstpackages.targets

```diff
@@ -0,0 +1,181 @@
+    
+    <PropertyGroup>
+        <!-- Use the alternative PackageReferenceName metadata if available -->
+        <_InjectProjectReferenceDependencyName Condition="'$(_PackageReferenceName)' == ''">%(_PackageProjectReference.FileName)</_InjectProjectReferenceDependencyName>
+        <_InjectProjectReferenceDependencyName Condition="'$(_PackageReferenceName)' != ''">$(_PackageReferenceName)</_InjectProjectReferenceDependencyName>
+    </PropertyGroup>
+    <ItemGroup>
+      <_InjectProjectReferenceDependency  Include="$(_InjectProjectReferenceDependencyName)">
+        <Name>$(_InjectProjectReferenceDependencyName)</Name>
+        <OriginalName>$(_InjectProjectReference)</OriginalName>
+        <Version>$(_AssemblyVersion)</Version>
+        <FilePath>%(_PackageProjectReference.Identity)</FilePath>
+        <MSBuildSourceProjectFile>%(_PackageProjectReference.MSBuildSourceProjectFile)</MSBuildSourceProjectFile>
+        <TargetGroup>%(_PackageProjectReference.TargetGroup)</TargetGroup>
+      </_InjectProjectReferenceDependency>
+    </ItemGroup>
+
+    <Message Condition="'@(_InjectProjectReferenceDependency)' != ''" Text="Converting project reference (%(_InjectProjectReferenceDependency.OriginalName)) to package dependency - %(_InjectProjectReferenceDependency.Name): %(_InjectProjectReferenceDependency.Version)" />
+    
+    <PropertyGroup>
+       <ErrorNoVersion Condition="'@(_InjectProjectReferenceDependency)' != '' and '$(_AssemblyVersion)' == ''">true</ErrorNoVersion>
+    </PropertyGroup>
+    <Error Condition="'$(ErrorNoVersion)' == 'true'" Text="Unable to determine version number for project reference dependency '%(_InjectProjectReferenceDependency.Name)'" />
+  </Target>
+  
+  <Target Name="AddDependenciesToProjectJson"
+          DependsOnTargets="GatherProjectReferenceForProjectJson;DetermineProjectJsonPath"
+          Condition="'@(_InjectProjectReferenceDependency)' != ''">
+    
+    <Message Condition="'$(PackageBuildNumberOverride)' != ''" Text="Setting package version to '$(PackageBuildNumberOverride)'" />
+    <GetPackageNumberFromPackageDrop Condition="'@(_PackagesDrops)' != ''"
+                                     PackageDrops="@(_PackagesDrops)">
+      <Output TaskParameter="PackageNumber" PropertyName="_DerivedPackageNumber" />
+    </GetPackageNumberFromPackageDrop>
+    
+    <Message Condition="'$(PackageBuildNumberOverride)' == '' and '$(_DerivedPackageNumber)' != ''" Text="Derived package version from packagedrops.  Setting package version to '$(_DerivedPackageNumber)'" />
+    <PropertyGroup>
+        <PackageBuildNumberOverride Condition="'$(PackageBuildNumberOverride)' == '' and '$(_DerivedPackageNumber)' != ''">$(_DerivedPackageNumber)</PackageBuildNumberOverride>
+        <VersionStructureRegex>(\d+\.\d+\.\d+)-((beta|rc2|rc3)-?(\d+)?)</VersionStructureRegex>
+        <BuildNumberOverrideStructureRegex>([^-]+)?-?(\d+(-\d\d)?)?</BuildNumberOverrideStructureRegex>
```

Comment:
All these regex's in the targets file kind of scare me (less so then having them hard coded in the task library but they still scare me. 

---
In reply to: [58437772](https://github.com/dotnet/buildtools/pull/571#discussion_r58437772) [](ancestors = 58437772)